transcom/mymove
# B 20173 int

## [B-20173](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A961676)

## Summary
This PR adds a new column into the moving_expenses table called sit_location and updates the moving-expenses API calls in Internal and GHC to use the sit_location field. This is the another in a series of stories that will give Customers the ability to add amount of weight stored and specify sit location for a moving expense on their PPM shipment and allow Office users to see the the fields on the Review Documents page. 

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Login as a Customer and create and submit a new PPM shipment
2. Login as a Service Counselor and approve the PPM shipment
3. Login as the Customer and click the Upload PPM Documents button
4. Enter the data on the first page click Save & Continue
5. Add a new moving expense using type of "Storage"
6. Inspect in the Chrome dev tools and look at the Network tab and mto_shipments
7. Expand the shipment and look for moving expenses and expand your moving expense
8. You should see a ppm shipment id, the moving expense id (listed as just "id") and the eTag
9. Copy these values off to use in swagger calls
10. Open [swagger for Internal ](http://milmovelocal:3000/swagger-ui/internal.html#/ppm/updateMovingExpense)
11. Enter your copied values for the updateMovingExpense PATCH call
12. Enter sitLocation in the body "ORIGIN" or "DESTINATION", something like this:   "sitLocation": "ORIGIN"
13. Update the "amount" to be greater than 1
14. **IMPORTANT**: Make sure you change the default: "movingExpenseType": "CONTRACTED_EXPENSE" to use 
**"movingExpenseType": "STORAGE",**
 * The weight_stored and sitLocation will only update for STORAGE type moving expenses. You can verify it only does update them by leaving "CONTRACTED_EXPENSE" and entering values for them, the sitLocation and weightStored should be null once you execute.
15. Execute the call and verify that you get a success message and see your sitLocation value
16. Optionally also look for your moving expense in the moving_expense database table to verify you have a sitLocation value
17. Go back to your Customer page and pull the new eTag value
18. Open the [GHC swagger page](http://officelocal:3000/swagger-ui/ghc.html#/ppm/updateMovingExpense)
19. Enter your values for the PATCH call and a different sitLocation amount to the body:   "sitLocation": "DESTINATION"
20. Execute and verify success and check DB table
21. Go to the [swagger for Internal ](http://milmovelocal:3000/swagger-ui/internal.html#/ppm/updateMovingExpense)
22. Scroll to the POST call to create a moving expense
23. Enter your ppm shipment id and execute
24. Verify the create was successful

### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
